src/main/java/uk/gov/hmcts/reform/roleassignment/launchdarkly/LDFlagChecker.java

```diff
@@ -5,44 +5,54 @@
 import com.launchdarkly.sdk.LDUser;
 import com.launchdarkly.sdk.server.LDClient;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.CacheConfig;
+import org.springframework.cache.annotation.CacheEvict;
 import org.springframework.cache.annotation.Cacheable;
+import org.springframework.cache.annotation.EnableCaching;
 import org.springframework.context.annotation.Configuration;
-import uk.gov.hmcts.reform.roleassignment.util.Constants;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
 
-@Cacheable(cacheNames = "ldconfig")
+@EnableCaching
+@CacheConfig(cacheNames = {"ldconfig" })
+@EnableScheduling
 @Configuration
 @Slf4j
 public class LDFlagChecker {
 
     @Value("${launchdarkly.sdk.environment}")
     private String environment;
 
-    @Value("${launchdarkly.sdk.testkey}")
-    private String sdkTestKey;
+    @Autowired
+    CacheManager cacheManager;
 
-    @Value("${launchdarkly.sdk.prodkey}")
-    private String sdkProdKey;
+    @Autowired
+    private LDConfiguration ldConfiguration;
 
+    @Cacheable({ "ldconfig"})
     public boolean verifyServiceAndFlag(String serviceName, String flagName) throws IOException {
-        String sdkKey;
-        if (environment.equalsIgnoreCase(Constants.AAT) || environment.equalsIgnoreCase(Constants.PROD)) {
-            sdkKey = sdkProdKey;
-        } else {
-            sdkKey = sdkTestKey;
-        }
-
-        LDClient client = new LDClient(sdkKey);
 
         LDUser user = new LDUser.Builder(environment)
             .firstName(environment)
             .lastName("user")
             .custom("servicename", serviceName)
             .build();
 
-        boolean showFeature = client.boolVariation(flagName, user, false);
+        boolean showFeature = ldConfiguration.getLdClient().boolVariation(flagName, user, false);
 
-        client.close();
+        ldConfiguration.getLdClient().close();
         return showFeature;
     }
+
+    @Scheduled(fixedRate = 6000)
+    @CacheEvict(allEntries = true, value = "ldconfig")
+    public void evictAllCaches() {
+        log.info("cleaning caches");
+        cacheManager.getCacheNames()
+                    .parallelStream()
```

Comment:
Is parallel steam required here? there would be only one cache object for this implementation.

Reply:
Removed the whole code. It's no longer required.